bundled/tool/ruff_server.py

```diff
@@ -0,0 +1,89 @@
+import os
+import shutil
+import site
+import subprocess
+import sys
+import sysconfig
+from pathlib import Path
+
+RUFF_EXE = "ruff.exe" if sys.platform == "win32" else "ruff"
+
+BUNDLE_DIR = Path(__file__).parent.parent
+
+
+def update_sys_path(path_to_add: str) -> None:
+    """Add given path to `sys.path`."""
+    if os.path.isdir(path_to_add):
+        # The `site` module adds the directory at the end, if not yet present; we want
+        # it to be at the beginning, so that it takes precedence over any other
+        # installed versions.
+        sys.path.insert(0, path_to_add)
+
+        # Allow development versions of libraries to be imported.
+        site.addsitedir(path_to_add)
+
+
+if __name__ == "__main__":
+    # Ensure that we can import bundled libraries like `packaging`
+    update_sys_path(os.fspath(BUNDLE_DIR / "libs"))
+
+
+from packaging.specifiers import SpecifierSet
+from packaging.version import Version
+
+# This is the first release that included `ruff server`.
+# The minimum version may change in the future.
+RUFF_VERSION_REQUIREMENT = SpecifierSet(">=0.3.3")
```

Comment:
For simplicity, should we just make this `0.3.5` or later?

Reply:
I thought it would be awkward to implement this feature if we only supported the latest version 😅 

But we probably should just make that the minimum, yeah.

Reply:
We'll also have to bump the bundled Ruff version to make this work (it's at `0.3.3` right now)

Reply:
That's fine, let me know if you need help doing so.